src/fast_check/transform_dts.rs

```diff
@@ -504,66 +504,7 @@ impl<'a> FastCheckDtsTransformer<'a> {
         fn_decl.function.body = None;
         fn_decl.declare = is_declare;
 
-        for param in &mut fn_decl.function.params {
-          match &mut param.pat {
-            Pat::Ident(ident) => {
-              if ident.type_ann.is_none() {
-                self.mark_diagnostic_any_fallback(ident.range());
-                ident.type_ann = Some(any_type_ann());
-              }
-            }
-            Pat::Assign(assign_pat) => {
-              match &mut *assign_pat.left {
-                Pat::Ident(ident) => {
-                  if ident.type_ann.is_none() {
-                    ident.type_ann = self.infer_expr_fallback_any(
-                      *assign_pat.right.clone(),
-                      false,
-                      false,
-                    );
-                  }
-
-                  ident.optional = true;
-                  param.pat = Pat::Ident(ident.clone());
-                }
-                Pat::Array(arr_pat) => {
-                  if arr_pat.type_ann.is_none() {
-                    arr_pat.type_ann = self.infer_expr_fallback_any(
-                      *assign_pat.right.clone(),
-                      false,
-                      false,
-                    );
-                  }
-
-                  arr_pat.optional = true;
-                  param.pat = Pat::Array(arr_pat.clone());
-                }
-                Pat::Object(obj_pat) => {
-                  if obj_pat.type_ann.is_none() {
-                    obj_pat.type_ann = self.infer_expr_fallback_any(
-                      *assign_pat.right.clone(),
-                      false,
-                      false,
-                    );
-                  }
-
-                  obj_pat.optional = true;
-                  param.pat = Pat::Object(obj_pat.clone());
-                }
-                Pat::Rest(_)
-                | Pat::Assign(_)
-                | Pat::Expr(_)
-                | Pat::Invalid(_) => {}
-              };
-            }
-            Pat::Array(_)
-            | Pat::Rest(_)
-            | Pat::Object(_)
-            | Pat::Invalid(_)
-            | Pat::Expr(_) => {}
-          }
-        }
-
+        self.handle_func_params(&mut fn_decl.function.params);
```

Comment:
This PR was mostly just extracting functions and then reusing the code in methods/ctors. I didn't change the internal implementations.